api4/license_test.go

```diff
@@ -75,6 +81,85 @@ func TestUploadLicenseFile(t *testing.T) {
 		CheckBadRequestStatus(t, resp)
 		require.False(t, ok)
 	})
+
+	t.Run("server has already gone through trial", func(t *testing.T) {
+		th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ExperimentalSettings.RestrictSystemAdmin = false })
+		mockLicenseValidator := mocks2.LicenseValidatorIface{}
+		defer testutils.ResetLicenseValidator()
+
+		//startTimestamp, err := time.Parse("2 Jan 2006 3:04 pm", "1 Jan 2021 12:00 am")
+		//require.Nil(t, err)
+
+		userCount := 100
+		mills := model.GetMillis()
+
+		license := model.License{
+			Id: "AAAAAAAAAAAAAAAAAAAAAAAAAA",
+			Features: &model.Features{
+				Users: &userCount,
+			},
+			Customer: &model.Customer{
+				Name: "Test",
+			},
+			StartsAt:  mills + 100,
+			ExpiresAt: mills + 100 + 2534400000 + 1,
+		}
+
+		mockLicenseValidator.On("LicenseFromBytes", mock.Anything).Return(&license, nil).Once()
+		licenseBytes, _ := json.Marshal(license)
+		licenseStr := string(licenseBytes)
+
+		mockLicenseValidator.On("ValidateLicense", mock.Anything).Return(true, licenseStr).Once()
+		utils.LicenseValidator = &mockLicenseValidator
+
+		licenseManagerMock := &mocks.LicenseInterface{}
+		licenseManagerMock.On("CanStartTrial").Return(false, nil).Once()
+		th.App.Srv().LicenseManager = licenseManagerMock
+
+		ok, resp := th.SystemAdminClient.UploadLicenseFile([]byte("sadasdasdasdasdasdsa"))
+		require.False(t, ok)
+		require.Equal(t, http.StatusBadRequest, resp.StatusCode)
+		require.Equal(t, "api.license.request-trial.can-start-trial.not-allowed", resp.Error.Id)
+	})
+
+	t.Run("allow uploading sanctioned trials even if server already gone through trial", func(t *testing.T) {
+		mockLicenseValidator := mocks2.LicenseValidatorIface{}
+		defer testutils.ResetLicenseValidator()
+
+		userCount := 100
+		mills := model.GetMillis()
+
+		license := model.License{
+			Id: "PPPPPPPPPPPPPPPPPPPPPPPPPP",
+			Features: &model.Features{
+				Users: &userCount,
+			},
+			Customer: &model.Customer{
+				Name: "Test",
+			},
+			IsTrial:   true,
+			StartsAt:  mills + 100,
+			ExpiresAt: mills + 100 + 2534400000,
```

Comment:
Add comment or define this number, so reader knows what it is?

Reply:
Done.